src/Components/Components/src/ElementRef.cs

```diff
@@ -1,22 +1,27 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using Microsoft.JSInterop.Internal;
 using System;
-using System.Collections.Generic;
+using System.Text.Json.Serialization;
 using System.Threading;
 
 namespace Microsoft.AspNetCore.Components
 {
     /// <summary>
     /// Represents a reference to a rendered element.
     /// </summary>
-    public readonly struct ElementRef : ICustomArgSerializer
+    public sealed class ElementRef
```

Comment:
Could you clarify why this is now becoming a `class`?

Reply:
As of when I tried, System.Text.Json did not support serializing value types. I think it's ongoing work.

Reply:
Since this is part of the public API, the class-or-structness is going get fixed forever once we ship 3.0. It would be very sad if we ended up doing unnecessary allocations for every `ElementRef` forever!

Do you know if `System.Text.Json` is tracking fixing this on a particular timescale?

Could we have an issue in our repo to track reverting this on a suitable milestone?

Reply:
It's currently out of scope for 3.0. https://github.com/dotnet/corefx/issues/37434. I'll file a tracking issue for this - maybe if we could make this work with a custom converter.

Reply:
This is back to being a reference type now.

Reply:
You can write a custom value converter which can be removed when the serializer supports constructors.